### PR TITLE
Fix hatchling build by adding required sections

### PR DESCRIPTION
**Pull request by swernli**
This adds the newly required sections to the hatchling build and making sure the right files get included in the wheel.

See https://github.com/pypa/hatch/issues/1113 for context.

### CODE REVIEW

**Review by idavis — APPROVED** [review]

File: jupyterlab/pyproject.toml [hunk 1]
```diff
@@ -60,5 +60,11 @@ npm = ["jlpm"]
 source_dir = "src"
 build_dir = "qsharp-jupyterlab/labextension"
 
+[tool.hatch.build.targets.wheel]
+packages = ["qsharp-jupyterlab"]
+
+[tool.hatch.build.targets.wheel.force-include]
```
Comment (ofek) [inline jupyterlab/pyproject.toml:66]: Do you want this in the wheel or source distribution?

File: jupyterlab/pyproject.toml [hunk 2]
```diff
@@ -60,5 +60,11 @@ npm = ["jlpm"]
 source_dir = "src"
 build_dir = "qsharp-jupyterlab/labextension"
 
+[tool.hatch.build.targets.wheel]
+packages = ["qsharp-jupyterlab"]
```
Comment (ofek) [inline jupyterlab/pyproject.toml:64]: This directory does not exist, are you sure this is correct?

Comment (swernli) [reply jupyterlab/pyproject.toml:64]: that folder is a child folder under jupyterlab, and from testing locally it looks like it pulls in the right contents.

File: jupyterlab/pyproject.toml (hunk 1 above)
Comment (swernli) [reply jupyterlab/pyproject.toml:66]: It was in the wheel previously, we don't do an sdist for this but I wanted to make sure the contents were the same as before.

File: jupyterlab/pyproject.toml (hunk 2 above)
Comment (ofek) [reply jupyterlab/pyproject.toml:64]: Sorry to bother here, I'm just trying to help! Which of the following mappings do you want the wheel to use as a top-level directory?

1. `jupyterlab/qsharp-jupyterlab` ->`qsharp-jupyterlab` (`packages = ["jupyterlab/qsharp-jupyterlab"]`)
2. `jupyterlab` ->`jupyterlab` (`packages = ["jupyterlab"]`)

Comment (swernli) [reply jupyterlab/pyproject.toml:64]: when I unzip the wheel, the contents look like:
image
this matches the contents of the wheel that we have before the upgrade to hatch 1.19, so I think this is intentional.

Comment (ofek) [reply jupyterlab/pyproject.toml:64]: Okay thank you! I would strongly recommend the following config if it produces the contents that you expect:

```toml
[tool.hatch.build.targets.wheel]
packages = ["jupyterlab/qsharp-jupyterlab"]
```

Comment (ofek) [reply jupyterlab/pyproject.toml:64]: Oh I see what's happening, there are multiple project files in this repository and I was not looking at where it was located. My apologies! Everything looks great

**Review by ofek — APPROVED** [review]

**Pull request merged**